Source/CarthageKit/Project.swift

```diff
@@ -413,24 +413,52 @@ public final class Project { // swiftlint:disable:this type_body_length
 			.map(ResolvedCartfile.init)
 	}
 
+	/// Attempts to determine the latest version (whether satisfiable or not)
+	/// of the project's Carthage dependencies.
+	///
+	/// This will fetch dependency repositories as necessary, but will not check
+	/// them out into the project's working directory.
+	private func latestDependencies() -> SignalProducer<[Dependency: PinnedVersion], CarthageError> {
+		return loadResolvedCartfile()
+			.flatMap(.merge) { cartfile -> SignalProducer<(Dependency, PinnedVersion), CarthageError> in
+				return SignalProducer(cartfile.dependencies.keys)
+					.flatMap(FlattenStrategy.merge) { dependency -> SignalProducer<(Dependency, PinnedVersion), CarthageError> in
```

Comment:
This `flatMap` looks like it could be moved out a level, instead of being nested inside the `flatMap`.

Also, this could be `.merge` instead of `FlattenStrategy.merge`.